website/docs/blueprints/ai-ml/bionemo.md

```diff
@@ -0,0 +1,225 @@
+```
+
+Install Kubeflow Training Operator:
+```bash
+kubectl apply -k "github.com/kubeflow/training-operator/manifests/overlays/standalone?ref=v1.7.0"
+```
+
+First, lets verify that we have worker nodes running in the cluster.
+
+```bash
+kubectl get nodes
+```
+:::info
+```bash
+NAME                                          STATUS   ROLES    AGE   VERSION
+ip-100-64-243-22.us-west-2.compute.internal   Ready    <none>   47h   v1.29.0-eks-5e0fdde
+ip-100-64-70-74.us-west-2.compute.internal    Ready    <none>   47h   v1.29.0-eks-5e0fdde
+```
+:::
+
+Next, lets verify all the pods are running.
+
+```bash
+kubectl get pods -A
+```
+Make sure training-operator, nvidia-device-plugin and fsx-csi-controller pods are running and healthy.
+
+#### Run bionemo jobs
+
+Once we are sure everything is working, we can start submitting jobs to our clusters.
+
+First job we are going to run is uniref50-job which will download and split the data for efficient processing.
+
+```bash
+cd training
+kubectl apply -f uniref50-job.yaml
+```
+
+If you look at the yaml manifest you would see it is a kubernetes Job, which simply runs a python script within the bionemo container.
+
+This process takes relatively a long time (~50-60 hours). In order to see if it is working check the logs from this pod:
+
+```bash
+kubectl logs uniref50-download-xnz42
+[NeMo I 2024-02-26 23:02:20 preprocess:289] Download and preprocess of UniRef50 data does not currently use GPU. Workstation or CPU-only instance recommended.
+[NeMo I 2024-02-26 23:02:20 preprocess:115] Data processing can take an hour or more depending on system resources.
+[NeMo I 2024-02-26 23:02:20 preprocess:117] Downloading file from https://ftp.uniprot.org/pub/databases/uniprot/uniref/uniref50/uniref50.fasta.gz...
+[NeMo I 2024-02-26 23:02:20 preprocess:75] Downloading file to /fsx/raw/uniref50.fasta.gz...
+[NeMo I 2024-02-26 23:08:33 preprocess:89] Extracting file to /fsx/raw/uniref50.fasta...
+[NeMo I 2024-02-26 23:12:46 preprocess:311] UniRef50 data processing complete.
+[NeMo I 2024-02-26 23:12:46 preprocess:313] Indexing UniRef50 dataset.
+[NeMo I 2024-02-26 23:16:21 preprocess:319] Writing processed dataset files to /fsx/processed...
+[NeMo I 2024-02-26 23:16:21 preprocess:255] Creating train split...
+```
+
+After this job is done, our processed dataset will be placed in /fsx/processed directory.
+
+After this step we can run the training job by running:
+
+```bash
+cd training
+kubectl apply -f esm1nv_pretrain-job.yaml
+```
+
+This manifest uses kubeflow's Pytorch training CRD. There are bunch of parameters within this manifest that could be modified. See [Bionemo's Docs](https://docs.nvidia.com/bionemo-framework/latest/notebooks/model_training_esm1nv.html) for detailed explanations of each parameter and how to tune them.
+
+According to Training Operator's documentation, ...worker-0 is the main process. In order to check the process:
```

Comment:
Write some conclusions for the document. Who can benefit from this blueprint? Also, provide some details about why you chose P3 instances and what P3 offers compared to other GPU instances.